docs-src/dev-guide/sdk-features.md

```diff
@@ -0,0 +1,571 @@
+| [Interceptors](#interceptors) | Yes | Yes  | Yes        | Yes    | Yes  | No  |
+
+Interceptors are a mechanism for modifying inbound and outbound SDK calls.
+Interceptors are commonly used to add tracing and authorization to the scheduling and execution of Workflows and Activities.
+You can compare these to "middleware" in other frameworks.
+
+The TypeScript SDK comes with an optional interceptor package that adds tracing with [OpenTelemetry](https://www.npmjs.com/package/@temporalio/interceptors-opentelemetry).
+See how to use it in the [interceptors-opentelemetry](https://github.com/temporalio/samples-typescript/tree/main/interceptors-opentelemetry) code sample.
+
+The [Go SDK](https://github.com/temporalio/sdk-go) provides support for distributed tracing with **_Interceptors_**.
+Interceptors uses Temporal headers to create a call graph of a [Workflow](/workflows#), along with its [Activities](/activities#) and [Child Workflows](/workflows#child-workflow).
+
+There are several tracing implementations supported by the Temporal Go SDK.
+For an [OpenTracing](https://pkg.go.dev/go.temporal.io/sdk/contrib/opentracing) Interceptor, use `opentracing.NewInterceptor(opentracing.TracerOptions{})` to create a `TracingInterceptor`.
+
+## Message passing
+
+All SDKs support the ability to send Signal and Query Workflows.
+
+| Feature                     | Go  | Java | TypeScript | Python | .Net | PHP |
+| --------------------------- | --- | ---- | ---------- | ------ | ---- | --- |
+| [Signals](#signals)         | Yes | Yes  | Yes        | Yes    | Yes  | Yes |
+| [Queries](#queries)         | Yes | Yes  | Yes        | Yes    | Yes  | Yes |
+| [Updates](#workflow-update) | Yes | Yes  | No         | No     | No   | No  |
+
+### Signals
+
+A Signal is an asynchronous request to a [Workflow Execution](/workflows#workflow-execution).
+
+### Queries
+
+A Query is a synchronous operation that is used to get the state of a [Workflow Execution](/workflows#workflow-execution).
+The state of a running Workflow Execution is constantly changing.
+You can use Queries to expose the internal Workflow Execution state to the external world.
+Queries are available for running or completed Workflows Executions only if the Worker is up and listening on the Task Queue.
+
+### Workflow Update
+
+An Workflow Update is a request to and a response from a Temporal Client to a Workflow Execution.
+
+The Workflow must have a function to handle the Update.
+Unlike a Signal handler, the Update handler function can mutate the state of the Workflow while also returning a value to the caller.
+The Update handler listens for Updates by the Update's name.
+
+When there is the potential for multiple Updates to cause a duplication problem, Temporal recommends adding idempotency logic to your Update handler that checks for duplicates.
+
+For more information, see [What is an Update?](/workflows#update).
+
+## Remote codec
```

Comment:
Remote codec is simply a protocol not a feature of an SDK. They are features of the UI and CLI. Some SDKs that have "Yes" below have no "remote codec" utilities or anything in the SDK at all about remote codecs.